data-prepper-api/src/main/java/org/opensearch/dataprepper/model/sink/AbstractSink.java

```diff
@@ -17,14 +17,25 @@
  * This class implements the Sink interface and records boilerplate metrics
  */
 public abstract class AbstractSink<T extends Record<?>> implements Sink<T> {
+    static private final String NUM_RETRIES = "num_retries";
+    static private final int NUM_DEFAULT_RETRIES = 600;
     protected final PluginMetrics pluginMetrics;
     private final Counter recordsInCounter;
     private final Timer timeElapsedTimer;
+    private final Thread retryThread;
 
     public AbstractSink(final PluginSetting pluginSetting) {
         this.pluginMetrics = PluginMetrics.fromPluginSetting(pluginSetting);
         recordsInCounter = pluginMetrics.counter(MetricNames.RECORDS_IN);
         timeElapsedTimer = pluginMetrics.timer(MetricNames.TIME_ELAPSED);
+        Integer numRetries = (Integer) pluginSetting.getAttributeFromSettings(NUM_RETRIES);
```

Comment:
I tend to think that Data Prepper core should be responsible for determining the number of allowed retries, or a global time to success.

If we do want to stick with having the sinks configure them, we should still have each sink configuration provide the setting and then pass it into the `AbstractSink` class. We are removing `PluginSetting` in favor or plugin configuration classes. The current work is somewhat incompatible with that upcoming change.

Reply:
I am OK with setting it to fixed value in the AbstractSink class. Shall we go with that approach?